src/zip_node.js

```diff
@@ -69,7 +70,7 @@ const zipNodeJsWithEsbuild = async function ({
     const { archive, output } = startZip(destPath)
     const { srcFile, stat } = await addStat(bundledFilePath)
 
-    addEntryFile(destFolder, archive, filename, bundledFilePath)
+    addEntryFile(destFolder, archive, jsFilename, bundledFilePath)
```

Comment:
Should `addEntryFile()` only use `bundleFilePath` instead of `jsFilename`? Since this is the file actually being required.
This would also remove the need to fix the file extension, providing esbuild always outputs a `.js` file.

Reply:
`bundleFilePath` will contain a full path, whereas `jsFilename` contains the relative path, which is what we want to include in the entry file. Or am I misunderstanding?

Reply:
> This would also remove the need to fix the file extension, providing esbuild always outputs a .js file.

You actually need to provide an output file with a `.js` extension so that it builds for the right target. If the output file is `.ts`, it outputs TypeScript.

(There may be an option to modify this behaviour I don't know about.)

Reply:
Makes sense! :+1: